app/controllers/reviews_controller.rb

```diff
@@ -12,6 +12,12 @@ def index
 
     @reviews_map[nil] = @reviews.where(semester_id: nil).sort_by{|r| r.course.mnemonic_number}
 
+    average_prof_rating = 0
+    @reviews.each do |r|
+      average_prof_rating += r.overall
+    end
+    @avg_prof_rating = (average_prof_rating/@reviews.count).round(2)
```

Comment:
I think in this case since the 'filter' is getting the current user's reviews, which is already done in this controller's index function, it seems to make more sense in the controller.